sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -31,15 +31,177 @@
+        string_type_columns = set(col_name for typ, col_name in zip(df.data, df.columns)
+                                  if isinstance(typ, StringArrayType))
+
+        for typ, col_name in zip(other.data, other.columns):
+            if isinstance(typ, StringArrayType):
+                string_type_columns.add(col_name)
+
+        def get_dataframe_column(df, column, idx):
+            return f'new_col_{column}_data_{df} = get_dataframe_data({df}, {idx})'
+
+        def get_append_result(df1, df2, column):
+            s1 = f'init_series(new_col_{column}_data_{df1})'
+            s2 = f'init_series(new_col_{column}_data_{df2})'
+            return f'new_col_{column} = {s1}.append({s2})._data'
+
+        func_definition = [f'def sdc_pandas_dataframe_{_func_name}_impl({", ".join(func_args)}):']
+        func_text = []
+        column_list = []
+
+        func_text.append(f'len_df = len(get_dataframe_data(df, {0}))')
+        func_text.append(f'len_other = len(get_dataframe_data(other, {0}))')
+
+        for col_name, i in df_columns_indx.items():
+            func_text.append(get_dataframe_column('df', col_name, i))
+            if col_name in other_columns_indx:
+                func_text.append(get_dataframe_column('other', col_name, other_columns_indx.get(col_name)))
+                func_text.append(get_append_result('df', 'other', col_name))
+            else:
+                func_text.append(f'new_col_{col_name}_data = init_series(new_col_{col_name}_data_df)._data')
+                if col_name in string_type_columns:
+                    func_text.append(f'new_col_{col_name} = fill_str_array(new_col_{col_name}_data, len_df+len_other)')
+                else:
+                    func_text.append(f'new_col_{col_name} = fill_array(new_col_{col_name}_data, len_df+len_other)')
+            column_list.append((f'new_col_{col_name}', col_name))
+
+        for col_name, i in other_columns_indx.items():
+            if col_name not in df_columns_indx:
+                func_text.append(get_dataframe_column('other', col_name, i))
+                func_text.append(f'new_col_{col_name}_data = init_series(new_col_{col_name}_data_other)._data')
+                if col_name in string_type_columns:
+                    func_text.append(f'new_col_{col_name} = fill_str_array(new_col_{col_name}_data, len_df+len_other, push_back=False)')
+                else:
+                    func_text.append(f'new_col_{col_name} = '
+                                     f'fill_array(new_col_{col_name}_data, len_df+len_other, push_back=False)')
+                column_list.append((f'new_col_{col_name}', col_name))
+
+        data = ', '.join(column for column, _ in column_list)
+        # TODO: Handle index
+        index = None
+        col_names = ', '.join(f"'{column_name}'" for _, column_name in column_list)
+        func_text.append(f"return sdc.hiframes.pd_dataframe_ext.init_dataframe({data}, {index}, {col_names})\n")
```

Comment:
```suggestion
        func_text.append(f"return init_dataframe({data}, {index}, {col_names})\n")
```

And then pass `sdc.hiframes.pd_dataframe_ext.init_dataframe` as global into exec?